src/main/perl/Blockdevices.pm

```diff
@@ -465,4 +466,33 @@ sub has_filesystem
     return $f =~ m{\s$fsregex\s+file}i;
 }
 
+=pod
+
+=head2 get_uuid
+
+Fetches the (PART)UUID of the device with blkid. Returns undef when not found.
+if part is true, we use PARTUUID instead of UUID
+
+=cut
+
+sub get_uuid
+{
+    my ($self, $part) = @_;
+    my $device = $self->devpath; 
+    my ($uuid, $re);
+    my $output = CAF::Process->new([BLKID, $device], log => $this_app)->output();
+    if($part) {
+        $re = qr!\sPARTUUID="(\S+)"!m ;
+    } else {
+        $re = qr!\sUUID="(\S+)"!m ;
+    }
+    if($output && $output =~ m/$re/m){
+        $uuid = $1;
+    }
+    if (!defined($uuid)){
+        $this_app->warn("${part}UUID of device $device could not be found");
```

Comment:
the message really needs `$part = $part ? 'PART' : '';` if $part is truly a boolean